supervisor/poller.py

```diff
@@ -78,15 +81,28 @@ def initialize(self):
         self._poller = select.poll()
         self.READ = select.POLLIN | select.POLLPRI | select.POLLHUP
         self.WRITE = select.POLLOUT
+        self.readables = set()
+        self.writables = set()
 
     def register_readable(self, fd):
         self._poller.register(fd, self.READ)
+        self.readables.add(fd)
 
     def register_writable(self, fd):
         self._poller.register(fd, self.WRITE)
+        self.writables.add(fd)
 
-    def unregister(self, fd):
+    def unregister_readable(self, fd):
+        self.readables.discard(fd)
         self._poller.unregister(fd)
+        if fd in self.writables:
+            self._poller.register(fd, self.WRITE)
```

Comment:
In kqueue you can unregister a file descriptor for specifically read or write but, as far as I can tell, [you cannot do that with poll](https://docs.python.org/2/library/select.html#select.poll.unregister). The workaround is to just reregister for write if we unregistered read and vice versa.